server/app/controllers/obs.py

```diff
@@ -102,3 +120,84 @@ def human_readable_obs(obs: Observation) -> Observation:
     obs.tca = obs["tca"].strftime("%Y-%m-%d %H:%M:%S") + ", " + strfdelta(aos_tca_duration, fmt="{M:02}m {S:02}s since AOS")
     obs.los = obs["los"].strftime("%Y-%m-%d %H:%M:%S") + ", " + strfdelta(aos_los_duration, fmt="{M:02}m {S:02}s since AOS")
     return obs
+
+@app.route('/obs/delete/<obs_id>', methods=["GET", "POST"])
+def obs_delete(obs_id: ObservationId = None):
+
+    # First check if such an observation even exists.
+    repository = Repository()
+    observation = repository.read_observation(obs_id)
+    if observation is None:
+        return render_template('obs_delete.html', status="There is no observation %s" % obs_id, obs_id=obs_id)
+
+    # Second, check if the guy is logged in.
+    if not current_user.is_authenticated:
+        return render_template('obs_delete.html', status = "You are not logged in, you can't delete anything.", obs_id=obs_id)
+
+    # Ok, at least this guy is logged in. Let's check who he is.
+    user_id = current_user.get_id()
+
+    # Check if the current user is the owner of the station.
+    station = repository.read_station(observation["station_id"])
+    station_id = station['station_id']
+
+    owners = repository.station_owners(station_id)
+    owner = False
+    for o in owners:
+        if o['id'] == user_id:
+            owner = True
+            break
+
+    if not owner:
+        return render_template('obs_delete.html', status = "You are not the owner of station %s, you can't delete observation %s." % (station.name, obs_id), obs_id=obs_id)
+
+    # If you got that far, this means the guy is logged in, he's the owner and is deleting his own observation.
+
+    status = obs_delete_real(repository, obs_id)
+    return render_template('obs_delete.html', status = status, obs_id=obs_id)
+
+def obs_delete_real(repository: Repository, obs_id: ObservationId):
+
+    # Step 1: delete products
+    files = repository.read_observation_files(obs_id)
+
+    root = app.config["storage"]['image_root']
+
+    status = ""
+
+    for f in files:
+        path = os.path.join(root, f['filename'])
+        try:
+            os.remove(f['filename'])
+            status += " Product %s deleted successfully." % path
+        except FileNotFoundError:
+            status += " Product %s was not found." % path
+
+        try:
+            fname = os.path.join(root, "thumbs", f['filename'])
+            os.remove(fname)
+            status += " Thumbnail file %s deleted successfully." % fname
+        except FileNotFoundError:
+            status += " Thumbnail file %s was not found." % fname
+
+
+    # Step 2: delete pass charts
+    try:
+        f = os.path.join(root, "charts", "by_time-%s.png" % obs_id)
```

Comment:
We generate two charts, but you remove only one.

Reply:
Yup. The other one (`polar-X.png`) is deleted a bit further down, in lines 184-189.